### PR TITLE
Adding first set of API tests

### PR DESCRIPTION
**Pull request by dangood84**

### CODE REVIEW

File: CypressTests/package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "dansacademisationapicypresstests",
```
Comment (cshnimble) [inline CypressTests/package.json:2]: Worth renaming this now it's part of the academisation-api project

File: CypressTests/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
```
Comment (cshnimble) [inline CypressTests/package.json:6]: Worth updating the scripts section to include options for running Cypress tests.
e.g. see how [a2b internal](https://github.com/DFE-Digital/a2b-internal/blob/main/ApplyToBecomeInternal/ApplyToBecomeCypressTests/package.json#L7) does it

File: CypressTests/cypress/e2e/Get-UpdateApplication.cy.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+
+describe('Academisation API Testing', () => {
+
+    let apiKey = Cypress.env('apiKey');
+    let applicationNumber = 24
+    let emailRegex = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/
+    let getDateTimestampFormatRegex = /^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3})Z?$/
+   
+    Cypress.config('baseUrl', 'http://academies-academisation-api-dev.azurewebsites.net/')
```
Comment (cshnimble) [inline CypressTests/cypress/e2e/Get-UpdateApplication.cy.js:21]: This should ideally be an environment variable, as with API key above

File: CypressTests/cypress/e2e/Get-UpdateApplication.cy.js [hunk 4]
```diff
@@ -0,0 +1,269 @@
+/// <reference types="Cypress" />
+import {UnauthorisedUserCannotUpdatePayload} from '../support/payloads/UnauthorisedUserCannotUpdateBody.spec'
```
Comment (cshnimble) [inline CypressTests/cypress/e2e/Get-UpdateApplication.cy.js:2]: One for a future enhancement - as these payloads are very similar but with a few changed bits of data, could we instead have one import of a positive test payload and just modify the data within the test?

e.g.
import { goodPayload } from '/path/to/payload'
[...]
let somePayload = goodPayload         // uses the happy path payload
somePayload.someField = 'badData'  // now modify data for failure condition

**Comment by cshnimble** [comment]
I would also recommend updating the repo's README.md with some instructions on how to set up and run the Cypress tests - see [a2b internal](https://github.com/DFE-Digital/a2b-internal) for an example

**Review by cshnimble — CHANGES_REQUESTED** [review]

**Review by cshnimble — APPROVED** [review]

**Comment by cshnimble** [comment]
@dangood84 Are these tests still valid? If not they need updating for this PR - it's been sat stale a long time

**Comment by dangood84** [comment]
> 

Hey Chris, Should still be valid.  Have my changes merged in OK.  I saw the prompts in VS Code to merge and update for the tests to use Application ID 27.  Have these changes taken affect from my last push?

Many thanks

**Pull request merged**